internal/storage/image.go

```diff
@@ -873,3 +880,43 @@ func GetImageService(ctx context.Context, sc *types.SystemContext, store storage
 
 	return is, nil
 }
+
+// FilterPinnedImage checks if the give image needs to be pinned
+// to exclude from kubelet's image GC.
+func FilterPinnedImage(image string, config *config.Config) bool {
+	if len(config.PinnedImages) == 0 {
+		return false
+	}
+	regexps := compileRegexpsForPinnedImages(config.PinnedImages)
```

Comment:
these should be saved in the imageService structure instead of compiled each time we pull